shared/api/cluster.go

```diff
@@ -82,11 +82,11 @@ type ClusterPut struct {
 	// API extension: clustering_join
 	ServerAddress string `json:"server_address" yaml:"server_address"`
 
-	// The trust password of the cluster you're trying to join
+	// The cluster join token for the cluster you're trying to join
 	// Example: blah
 	//
 	// API extension: clustering_join
-	ClusterPassword string `json:"cluster_password" yaml:"cluster_password"`
+	ClusterToken string `json:"cluster_token" yaml:"cluster_token"`
```

Comment:
Why is this being moved from InitClusterPreseed into ClusterPut (I realise ClusterPut is embedded in InitClusterPreseed).

Reply:
I understood that this comes from here: https://github.com/canonical/lxd/pull/13567/commits/b185d390c48b85bca4bea601a2714ba376670f85#diff-fa47c056c3b0d0529a066e4b2c5ef4777424a0b63ba078399a4c5e584e092ab0L200

On cluster join the token was reused to be the password so when removing the password field the token is essentially the same.

Reply:
I see. But if we make this change, the newer API client won't be able to access older LXD servers using trust password, or even using a token in the password field right?

Reply:
Fixed. The clients now fallback to password if the upstream server doesn't have the `explicit_trust_token` extension.